### PR TITLE
Kubernetes: add --kube-reserved, tweak --eviction-hard

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Adds standard `--kube-reserved` kubelet runtime configuration values to protect against vm OOM

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
Kubernetes: add --kube-reserved, tweak --eviction-hard
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@pengzhisun @andyzhangx @karataliu @lachie83 @slack @khenidak  @brendanburns FYI for any thoughts around these proposed changes to default kubelet runtime configuration

File: docs/clusterdefinition.md [hunk 1]
```diff
@@ -173,7 +173,8 @@ Below is a list of kubelet options that acs-engine will configure by default:
 |"--cluster-domain"|"cluster.local"|
 |"--pod-infra-container-image"|"pause-amd64:*version*"|
 |"--max-pods"|"30", or "100" if using kubenet --network-plugin (i.e., `"networkPlugin": "kubenet"`)|
-|"--eviction-hard"|"memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%"|
+|"--eviction-hard"|"memory.available<5%,nodefs.available<10%,nodefs.inodesFree<5%"|
+|"--kube-reserved"|"cpu=1,ephemeral-storage=1Gi"|
```
Comment (feiskyer) [inline docs/clusterdefinition.md]: There is a reference in GKE docs: https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-architecture#node_allocatable

File: docs/clusterdefinition.md [hunk 2]
```diff
@@ -173,7 +173,8 @@ Below is a list of kubelet options that acs-engine will configure by default:
 |"--cluster-domain"|"cluster.local"|
 |"--pod-infra-container-image"|"pause-amd64:*version*"|
 |"--max-pods"|"30", or "100" if using kubenet --network-plugin (i.e., `"networkPlugin": "kubenet"`)|
-|"--eviction-hard"|"memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%"|
```
Comment (feiskyer) [inline docs/clusterdefinition.md]: It seems `100Mi` is used more often for memory, `memory.available<100Mi` is ok to me

**Comment by jackfrancis** [comment]
filed #3235 to do this the right way

**Pull request closed without merging**